.github/workflows/release.yml

```diff
@@ -0,0 +1,141 @@
+name: Release
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+  pull_request:
+    branches:
+      - main
+
+env:
+  # whether artifacts should be published to ci-job (previewed) or as release
+  preview: ${{ !startsWith(github.ref, 'refs/tags/') }}
+  
+jobs:
+  build:
+    name: Build
+    env:
+      CARGO: cargo  # 'cross' used for 32-bit and big-endian systems
+      RUST_BACKTRACE: 1  # emit backtrace on panics
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false # don't fail other jobs if one fails
+      matrix:
+        build: [x86_64-linux, x86_64-macos, x86_64-windows]
+        include:
+          - build: x86_64-linux
+            os: ubuntu-latest
+            rust: stable
+            target: x86_64-unknown-linux-musl
+            cross: false
+
+          - build: aarch64-linux
+            os: ubuntu-latest
+            rust: stable
+            target: aarch64-unknown-linux-gnu
+            cross: true
+
+          - build: riscv64-linux
+            os: ubuntu-latest
+            rust: stable
+            target: riscv64gc-unknown-linux-gnu
+            cross: true
```

Comment:
How about change these to musl too?

Reply:
To be honest, I don't have a good sense for the advantages of each. What would be the reasons to favor one over the other?

Reply:
For `riscv64`, it seems the toolchain to include `rust-std` isn't available yet for `musl`:
https://github.com/dgkf/R/actions/runs/6490904840/job/17627388865?pr=64#step:5:65

Reply:
Binaries built with `gnu` targets are not compatible with environments other than those in which they were built.
See, for example, the following problem with mdBook:
rust-lang/mdBook#1862
rust-lang/mdBook#1954

Reply:
I see - okay, then I'll add it where we can. It seems the `riscv64` is still going to have to stay on `gnu` (at least if we want to use the `stable` toolchain, not sure about nightly/others) but the others can be migrated.

Reply:
Thanks!